platform-grpc-service-framework/src/main/java/org/hypertrace/core/serviceframework/grpc/GrpcPlatformServiceContainer.java

```diff
@@ -199,6 +199,23 @@ public boolean healthCheck() {
     }
   }
 
+  private Duration[] generateLatencyHistogramBuckets() {
+    ArrayList<Duration> histogram_buckets = new ArrayList<>();
+    int factor = 3;
+    long curr_duration_ms =
+        getAppConfig().hasPath(MAX_LATENCY_HISTOGRAM_BUCKET_MS)
+            ? getAppConfig().getLong(MAX_LATENCY_HISTOGRAM_BUCKET_MS)
+            : DEFAULT_MAX_LATENCY_HISTOGRAM_BUCKET_MS;
+    while (curr_duration_ms >= 10) {
+      histogram_buckets.add(Duration.ofMillis(curr_duration_ms));
+      curr_duration_ms = curr_duration_ms / factor;
+      curr_duration_ms = curr_duration_ms - (curr_duration_ms % 5);
+    }
+    Duration[] histogram_buckets_array = new Duration[histogram_buckets.size()];
+    histogram_buckets_array = histogram_buckets.toArray(histogram_buckets_array);
+    return histogram_buckets_array;
```

Comment:
nit - the separated declarations don't lend any readability, can merge
```suggestion
    return histogram_buckets.toArray(new Duration[histogram_buckets.size()]);
```